app/controllers/devise/saml_sessions_controller.rb

```diff
@@ -65,18 +65,24 @@ def store_info_for_sp_initiated_logout
 
   # Override devise to send user to IdP logout for SLO
   def after_sign_out_path_for(_)
-    idp_entity_id = get_idp_entity_id(params)
-    request = OneLogin::RubySaml::Logoutrequest.new
-    saml_settings = saml_config(idp_entity_id).dup
+    # if user is signed out, verify_signed_out_user method will set a notice flash message
+    # based on that, will redirect user to after sign out path or create the SP initiated logout request
+    if flash[:notice].present?
```

Comment:
This is a good catch!

But using flash to determine this might end up being surprising, the flash can be used for other things. Maybe we should override the `#verify_signed_out_user` method, which would get rid of the need for a check here and avoid overriding the purpose of flash.

```ruby
  def verify_signed_out_user
    if all_signed_out?
      redirect_to Devise.saml_sign_out_success_url.presence || after_sign_out_path_for(resource_name)
    end
  end
```

Alternatively we could prepend our own method to add a designation that we could use in this check. Something like

```ruby
  prepend_before_action :check_signed_out_user, :verify_signed_out_user, :store_info_for_sp_initiated_logout, only: :destroy

  def check_signed_out_user
    @signed_out_user = all_signed_out?
  end
```

Then this would be updated to use that new attribute:

```suggestion
    if @signed_out_user
```

What do you think?

Reply:
@adamstegman Personally, I think the first solution is better. With the second, too many `before_action` actions is added, that may be confusing.